hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/validation/UsageLimits.java

```diff
@@ -168,26 +156,56 @@ public double percentAccountsUsed() {
         return 100.0 * numAccounts / dynamicProperties.maxNumAccounts();
     }
 
+    public int roundedAccountPercentUtil() {
+        final var n = dynamicProperties.maxNumAccounts();
+        return n == 0 ? 100 : (int) ((100 * updatedNumAccounts()) / n);
+    }
+
     public double percentContractsUsed() {
         return 100.0 * numContracts / dynamicProperties.maxNumContracts();
     }
 
+    public int roundedContractPercentUtil() {
+        final var n = dynamicProperties.maxNumContracts();
+        return n == 0 ? 100 : (int) ((100 * numContracts) / n);
```

Comment:
Good observation; since a contract is just a special kind of account, we have a different kind of book-keeping for `numContracts` that should always keep it up to date.